src/main/java/com/example/simulation/SimpleSimulation.java

```diff
@@ -58,18 +64,39 @@ public StepReport step() {
         }
         if (algorithm.isFinished()){
             removeSimulationRelatedColoring();
+            openResultDialog();
         }
         graphController.update();
         return report;
     }
 
+    public void openResultDialog() {
```

Comment:
I am not sure if it's the right place to call this method, because Simulation itself should be as separated from gui as possible. Did You consider moving this method to SinulationController? I think it would suit there better.

Reply:
moved